### DISCUSSION TITLE
[Feature] Add option to move pictures from one album to another (new or existing) album

### DISCUSSION THREAD

**Discussion by theone11** [immich-app/immich#8604] [Feature Request]
### The feature

I have pictures in an album and would like to move them out of the existing album and into a new **or** another existing album.

The current method I use:

- Mark the pictures and "Copy to album" where I choose an existing album or create a new one.
- Go back to the original album, find and mark the pictures **again** and remove them from the original album.

What I want to do
- Mark the pictures and "Move to album" where I choose an existing album or create a new one.

### Platform

- [ ] Server
- [X] Web
- [ ] Mobile

**Comment by soana** [comment]
This feature will save me a lot of clicks in organizing my albums. Right now I move all the pictures that are not in an album to a WIP album this way when opening the album in the Web GUI I can see dates and select my pictures using the shift buttons. if I try to do that from a search window  that includes only pictures not in an album there is no way to use shift to select a large group of pictures.
In photo prism the not in an album switch was present on the main page and it was easier to select and create albums than what immich has right now.

**Comment by skylerlink** [comment]
I would also find this very useful

**Comment by nerlins** [comment]
Seriously. Why is this not already implemented?? It's a waste of time having to add/ remove, instead of MOVE.

**Comment by L00maca** [comment]
Joining the request, it would make organizing albums sooooo much easier!

**Comment by Dunkelschunkel** [comment]
one up
timesaver it would be

**Comment by Gorbush** [comment]
This would be especially useful if you do like me - import from different sources like AmazonPhotos, Google Photos, iCloud, and some manual backups from different phones - and all of them were imported with option `create-albums` - which means each picture is actually present in 5 up to 10 different albums.

So some simple `Remove from all the albums` or an `X` next to the album name would be a huge time saver.

**Comment by mkubiak402** [comment]
Yeah how is this not a thing?? I would like to organize my pics and i cant because i cant move them to different album's

**Comment by rennerocha** [comment]
This is a very useful feature for the way that I use the application. I synchronize all my photos from my mobile into a "Camera" album, then I go manually moving photos to more specific albums. Currently I need to add to the other album and then remove from the "Camera" album. It takes much more time than just moving to another album.

The current "add to album and keep it in the current album" is useful as well, so I wouldn't remove this feature anyway. I use it when I want to select some photos from different albums to share with someone or to download them for printing. But this is a process that is done less frequently than the moving one.

**Comment by solarssk** [comment]
Hey. Any updates in this topic?

**Reply by bo0tzz** [reply]
https://justinmayer.com/posts/any-updates/

**Reply by solarssk** [reply]
you know, it's such a basic software feature - that I'm shocked that it's not implemented. From here it would be enough to write whether there are any plans to do this functionality or not. The post appeared in April, we are in July - if there is a product owner (yes I know, it is open source), it would be possible to write whether you want it at all or not. ;)

**Reply by bo0tzz** [reply]
If we didn't want it, the thread would be closed.

**Reply by solarssk** [reply]
well :) and great, it means that the topic is not burned (yet) and in some future perspective you can potentially count on this functionality. Thanks.

**Comment by boexler** [comment]
1+

**Comment by XobSod** [comment]
That would be nice to have, but I don't mid clicking tho.

I wanted just ask you @theone11 @rennerocha how are you even able to remove the photos from original album? Does it work for you? Because currently I have a situation that I want to move a photo from albumA to albumB:
- select photo on albumA -> Add to albumB
- select photo on albumA -> remove from (current) album

This theoretically works, the photo disappears from albumA and shows in albumB... until the phone backup runs and the photo gets readded to albumA.

So does this flow work for you?

**Comment by Zephyr750** [comment]
That would be very useful. My wife and I share an album. However, I have certain photos—for instance, of our home construction or tools—that I don't want included in that album. The ability to simply select these and *move* them to a different folder would be the absolute highlight of Immich for me. That way, I wouldn't end up with duplicates—as I do now—sitting in both the "Home Construction" folder and the "Album" folder.  I hope this feature is implemented soon.

**Comment by raj-magesh** [comment]
Found a couple of PRs attempting this that didn't get merged:

https://github.com/immich-app/immich/pull/18954 was rejected for concerns about lack of utility + overcrowding while https://github.com/immich-app/immich/pull/23518 was rejected for using some older internal APIs.

I've requested reopening the first one pointing to this discussion as evidence that the feature would be useful to multiple users.

**Comment by krozaniecki** [comment]
The function of transferring photos to an album in the web application of the mobile application would be very useful. I have 17,000 photos in my camera album that I would like to organize, but adding them to an album and then re-selecting them and deleting them from the album is terribly impractical and time-consuming :/ Of course, the number will increase, so it would be incredibly useful.

**Comment by kub3let** [comment]
For anyone coming here this is already possible just with a lot more clicks.

- Select your images from your custom Album
- Use the `+` (Add to Album), select your Album you want to move them to
- Then use the three dot menu -> Remove from Album

<img width="258" height="107" alt="image" src="https://github.com/user-attachments/assets/feda7aca-36b1-4d38-8eed-1a73d5fb8fc4" />

**Reply by solarssk** [reply]
That's not a solution - it's a workaround; it makes the “move from album to album” feature harder to use.

**Reply by krozaniecki** [reply]
+1 for @solarssk and besides that, in the android app you can't do as you say because after copying them to the album the photos are unchecked

**Reply by raj-magesh** [reply]
The issue is that, at scale, it's easy to mis-navigate away from the selection between steps 2 and 3, which leaves duplicate images that are then very hard to identify and remove---a "non-atomic" move.

Common operations need to be bulletproof.

**Reply by kub3let** [reply]
Not saying it's a great solution, but at least there is a way to do it currently.

**Reply by raj-magesh** [reply]
Yeah, definitely! Certainly helpful to share it for folks who didn't realize it's possible at all.

Don't want it to be used aa justification for the devs not to implement a more direct approach though. 😅

**Reply by rennerocha** [reply]
@kub3let it is possible to do it, and I believe that everyone knows that. But unfortunately it is not practical for more than a few photos. If you are moving photos from a travel for example, it is not really feasible to do that. Very prone to errors!

I am very surprised that #18954 was considered lack of utility given in this thread we have many people really asking that and providing very real use cases for that.
